src/duckling/entitysystem/services/entity-layer.service.ts

```diff
@@ -0,0 +1,84 @@
+import {
+    Component,
+    Input,
+    Output,
+    EventEmitter,
+    AfterViewInit,
+    Injectable
+} from '@angular/core';
+import {BehaviorSubject} from 'rxjs';
+
+import {Attribute, Entity, EntityKey} from '../entity';
+import {BaseAttributeService} from '../base-attribute.service';
+import {EntitySystemService} from '../entity-system.service';
+
+/**
+ * Function type that is used to set a position.
+ * @returns Returns the new value of the attribute.
+ */
+export type LayerGetter = (attribute : Attribute) => string;
+
+/**
+ * The EntityLayerService is used to retrieve the layer of an entity.
+ */
+@Injectable()
+export class EntityLayerService extends BaseAttributeService<LayerGetter> {
+
+    private _hiddenLayers : {[layerKey : string] : boolean} = {};
```

Comment:
This should be stored in the redux store. That allows us to implement hot reloading in the future. You can also use that to make layer changes undo-able. Look at the selection service as an example.